onnx/defs/tensor/defs.cc

```diff
@@ -3144,7 +3144,7 @@ ONNX_OPERATOR_SET_SCHEMA(
           auto* dim = output_shape.add_dim();
           if (hasInputShape(ctx, 0)) {
             const TensorShapeProto& input_shape = getInputShape(ctx, 0);
-            dim->set_dim_value(input_shape.dim_size());
+            dim->set_dim_value(input_shape.dim_size() == 0? 1 : input_shape.dim_size());
```

Comment:
Thanks for the investigation and I understand your concern. Another fact I would like to point out is even today np.nonzero is still behaving like `nonzero(atleast_1d(arr))` although there is a deprecation warning. So if ONNX decides not to behave the same as Current NumPy, the spec should clarify it to prevent confusion. Also, the test in ORT will need to be updated.